redis-persistence/src/main/java/com/netflix/conductor/dao/dynomite/RedisExecutionDAO.java

```diff
@@ -324,6 +324,11 @@ public boolean removeWorkflow(String workflowId) {
 			for (Task task : workflow.getTasks()) {
 				removeTask(task.getTaskId());
 			}
+
+			// After deleting tasks, remove the empty mapping sets
+			dynoClient.del(nsKey(WORKFLOW_TO_TASKS, workflow.getWorkflowId()));
+			dynoClient.del(nsKey(SCHEDULED_TASKS, workflow.getWorkflowId()));
```

Comment:
Can we delete these two before we delete the workflow?

Reply:
Moved both task deletion and mapping deletion before workflow deletion.